js/web/lib/wasm/jsep/webgpu/types.ts

```diff
@@ -189,6 +189,9 @@ export interface ComputeContext {
   output(index: number, dims: readonly number[]): number;
   getMaxComputeWorkgroupSizes(): [number, number, number];
   getMaxComputeWorkgroupStoragesize(): number;
+  isSubgroupsSupported(): boolean;
+  isSubgroupsF16Supported(): boolean;
+  getSubgroupSizeRange(): [number, number] | undefined;
```

Comment:
since those 5 functions always return const value (correct me if I'm wrong), how about create a type and put the value into it?

```ts
interface WebGpuDeviceInfo{
  readonly maxComputeWorkgroupSizes: readonly [number, number, number];
  readonly maxComputeWorkgroupStoragesize: number;
  readonly f16Supported: boolean;
  readonly subgroupsSupported: boolean;
  readonly subgroupsF16Supported: boolean;
  readonly subgroupSizeRange: readonly [number, number] | undefined;
}
```

Then add a `deviceInfo` in compute context ( now we have both deviceInfo and adapterInfo )

Reply:
Actually these five helper function is just a warper to access specific items of the `limits`/`features` field of the backend WebGPU `device`, all information are invariable for a given WebGPU `device` and already been kept in it.

Yes we can make a `WebGpuDeviceFeaturesAndLimits` etc. and copy the value into it (interested ones only?), but I thought that would be a duplication?

Reply:
Please take a look, thanks!

In the latest commit I make `DeviceInfoImpl` hold the actual values of subgroups info (just copied from `device.limits` and `device.features`), and remove the unused `maxComputeWorkgroupSizes` and `maxComputeWorkgroupStoragesize`.

On the other hand, currently no shader use the `subgroups` features and the related information, though.